File: app/src/main/java/com/snaptiongame/app/presentation/view/utils/ShowcaseUtils.java
package com.snaptiongame.app.presentation.view.utils;

import android.app.Activity;
import android.content.res.Resources;
import android.graphics.Point;
import android.support.v4.widget.NestedScrollView;
import android.view.View;

import com.snaptiongame.app.R;

import java.util.List;

import uk.co.deanwild.materialshowcaseview.MaterialShowcaseSequence;
import uk.co.deanwild.materialshowcaseview.MaterialShowcaseView;
import uk.co.deanwild.materialshowcaseview.ShowcaseConfig;

/**
 * @author Tyler Wong
 */

public class ShowcaseUtils {
    private static final int SHOWCASE_DELAY = 1000;

    public static void showShowcase(Activity activity, View target, int title, int content) {
        new MaterialShowcaseView.Builder(activity)
                .setTarget(target)
                .setDismissText(R.string.got_it)
                .setTitleText(title)
                .setContentText(content)
                .setDelay(SHOWCASE_DELAY)
                .singleUse(activity.getString(title))
                .show();
    }

    public static void showShowcaseSequence(Activity activity, NestedScrollView scrollView, List<View> targets,
                                            List<Integer> titles, List<Integer> contents) {
        Resources res = activity.getResources();
        ShowcaseConfig config = new ShowcaseConfig();
        config.setDelay(SHOWCASE_DELAY);

        MaterialShowcaseSequence sequence = new MaterialShowcaseSequence(
                activity, res.getString(titles.get(0)));
        sequence.setConfig(config);

        for (int index = 0; index < targets.size(); index++) {
            sequence.addSequenceItem(targets.get(index),
                    res.getString(titles.get(index)),
                    res.getString(contents.get(index)),
                    res.getString(R.string.got_it));
        }

        if (scrollView != null) {
            sequence.setOnItemDismissedListener((MaterialShowcaseView materialShowcaseView, int index) -> {
                if (index < targets.size() - 1) {
                    View nextView = targets.get(index + 1);
                    Point childOffset = new Point();
                    ViewUtils.getDeepChildOffset(scrollView, nextView.getParent(), nextView, childOffset);
                    scrollView.smoothScrollTo(0, childOffset.y);
                }
            });
        }

        sequence.start();
    }
}

File: app/src/main/java/com/snaptiongame/app/presentation/view/utils/ShowcaseUtils.kt
package com.snaptiongame.app.presentation.view.utils

import android.app.Activity
import android.graphics.Point
import android.support.v4.widget.NestedScrollView
import android.view.View

import com.snaptiongame.app.R

import uk.co.deanwild.materialshowcaseview.MaterialShowcaseSequence
import uk.co.deanwild.materialshowcaseview.MaterialShowcaseView
import uk.co.deanwild.materialshowcaseview.ShowcaseConfig

/**
 * @author Tyler Wong
 */

object ShowcaseUtils {
    private const val SHOWCASE_DELAY = 750

    @JvmStatic
    fun showShowcase(activity: Activity, target: View, title: Int, content: Int) {
        MaterialShowcaseView.Builder(activity)
                .setTarget(target)
                .setDismissText(R.string.got_it)
                .setTitleText(title)
                .setContentText(content)
                .setDelay(SHOWCASE_DELAY)
                .singleUse(activity.getString(title))
                .show()
    }

    @JvmStatic
    fun showShowcaseSequence(activity: Activity, scrollView: NestedScrollView?, targets: List<View>,
                             titles: List<Int>, contents: List<Int>) {
        val res = activity.resources
        val config = ShowcaseConfig()
        config.delay = SHOWCASE_DELAY.toLong()

        val sequence = MaterialShowcaseSequence(
                activity, res.getString(titles[titles.size - 1]))
        sequence.singleUse(res.getString(titles[titles.size - 1]))
        sequence.setConfig(config)

        for (index in targets.indices) {
            sequence.addSequenceItem(targets[index],
                    res.getString(titles[index]),
                    res.getString(contents[index]),
                    res.getString(R.string.got_it))
        }

        if (scrollView != null) {
            sequence.setOnItemDismissedListener { _: MaterialShowcaseView, index: Int ->
                if (index < targets.size - 1) {
                    val nextView = targets[index + 1]
                    val childOffset = Point()
                    ViewUtils.getDeepChildOffset(scrollView, nextView.parent, nextView, childOffset)
                    scrollView.smoothScrollTo(0, childOffset.y)
                }
            }
        }

        sequence.start()
    }
}
